flink-table/flink-table-planner/src/test/java/org/apache/flink/table/api/QueryOperationTestPrograms.java

```diff
@@ -0,0 +1,310 @@
+                                    .build())
+                    .runTableApi(t -> t.from("s").where($("a").isGreaterOrEqual(15)), "sink")
+                    .runSql(
+                            "SELECT * FROM (\n"
+                                    + "    SELECT `a`, `b` FROM `default_catalog`.`default_database`.`s`\n"
+                                    + ") WHERE `a` >= 15")
+                    .build();
+    static final TableTestProgram DISTINCT_QUERY_OPERATION =
+            TableTestProgram.of("distinct-query-operation", "verifies sql serialization")
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("s")
+                                    .addSchema("a bigint", "b string")
+                                    .producedValues(
+                                            Row.of(20L, "apple"),
+                                            Row.of(20L, "apple"),
+                                            Row.of(5L, "pear"))
+                                    .build())
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink")
+                                    .addSchema("a bigint", "b string")
+                                    .consumedValues(Row.of(20L, "apple"))
+                                    .build())
+                    .runTableApi(
+                            t -> t.from("s").where($("a").isGreaterOrEqual(15)).distinct(), "sink")
+                    .runSql(
+                            "SELECT DISTINCT * FROM (\n"
+                                    + "    SELECT * FROM (\n"
+                                    + "        SELECT `a`, `b` FROM `default_catalog`.`default_database`.`s`\n"
+                                    + "    ) WHERE `a` >= 15\n"
+                                    + ")")
+                    .build();
+    static final TableTestProgram AGGREGATE_QUERY_OPERATION =
+            TableTestProgram.of("aggregate-query-operation", "verifies sql serialization")
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("s")
+                                    .addSchema("a bigint", "b string")
+                                    .producedValues(
+                                            Row.of(10L, "apple"),
+                                            Row.of(20L, "apple"),
+                                            Row.of(5L, "pear"),
+                                            Row.of(15L, "pear"))
+                                    .build())
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink")
+                                    .addSchema("a string", "b bigint")
+                                    .consumedValues(
+                                            Row.ofKind(RowKind.INSERT, "apple", 10L),
+                                            Row.ofKind(RowKind.UPDATE_BEFORE, "apple", 10L),
```

Comment:
let's only define the materialized result here. we don't care about the changelog if this is not an ExecNode test, if we make the engine smarter later we need to update all these tests